lib/widgets/message_list.dart

```diff
@@ -511,60 +533,93 @@ class _UnreadMarker extends StatelessWidget {
   }
 }
 
-class StreamTopicRecipientHeader extends StatelessWidget {
-  const StreamTopicRecipientHeader({super.key, required this.message});
+class StreamMessageRecipientHeader extends StatelessWidget {
+  const StreamMessageRecipientHeader({super.key, required this.message, required this.showStream});
 
   final StreamMessage message;
+  final bool showStream;
 
   @override
   Widget build(BuildContext context) {
+    // For design specs, see:
+    //   https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=538%3A20849&mode=dev
+    //   https://github.com/zulip/zulip-mobile/issues/5511
     final store = PerAccountStoreWidget.of(context);
 
     final stream = store.streams[message.streamId];
-    final streamName = stream?.name ?? message.displayRecipient; // TODO(log) if missing
     final topic = message.subject;
-
-    final subscription = store.subscriptions[message.streamId];
-    final streamColor = Color(subscription?.color ?? 0x00c2c2c2);
+    final swatch = store.subscriptions[message.streamId]?.colorSwatch()
+      ?? _kFallbackStreamColorSwatch;
     final contrastingColor =
-      ThemeData.estimateBrightnessForColor(streamColor) == Brightness.dark
+      (ThemeData.estimateBrightnessForColor(swatch.barBackground) == Brightness.dark)
         ? Colors.white
         : Colors.black;
+    final textStyle = TextStyle(
+      color: contrastingColor,
+      fontFamily: 'Source Sans 3',
+      fontSize: 16,
+      letterSpacing: 0.02 * 16,
+      height: (18 / 16),
+    ).merge(weightVariableTextStyle(context, wght: 600, wghtIfPlatformRequestsBold: 900));
+
+    final streamWidget = (showStream)
+      ? GestureDetector(
+          onTap: () => Navigator.push(context,
+            MessageListPage.buildRoute(context: context,
+              narrow: StreamNarrow(message.streamId))),
+          child: Row(children: [
+            Padding(
+              // Figma shows 5px spacing but 6px here matches better visually.
+              padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 11),
+              child: Icon(size: 18, color: swatch.iconOnBarBackground,
+                // A null [Icon.icon] makes a blank space.
+                (stream != null) ? iconDataForStream(stream) : null)),
+            Padding(
+              padding: const EdgeInsets.symmetric(vertical: 11),
+              child: Text(stream?.name ?? message.displayRecipient, // TODO(log) if missing
+                style: textStyle,
+                maxLines: 1,
+                overflow: TextOverflow.ellipsis)),
+            Padding(
+              // Figma has 5px horizontal padding around an 8px wide icon.
+              // Icon is 16px wide here so horizontal padding is 1px.
+              padding: const EdgeInsets.symmetric(horizontal: 1, vertical: 12),
+              child: Icon(size: 16,
+                color: contrastingColor.withAlpha(154), // 60%
```

Comment:
> where the color shows up as `rgba(154, 154, 154, 1)` in `RGB` mode but `hsla(0, 0%, 60%, 1)` in `HSL`.

Wacky! Seems like Figma has a bug, probably multiplying by 256 instead of 255.

That really seems like the sort of thing that should be in Figma's wheelhouse. :-/